crates/proto/src/op/edns.rs

```diff
@@ -113,6 +118,11 @@ impl Edns {
     pub fn set_option(&mut self, option: EdnsOption) {
         self.options.insert(option);
     }
+
+    /// Remove the option specified by the `EdnsCode`
+    pub fn remove_option(&mut self, option: EdnsCode) {
+        self.options.remove(option);
+    }
```

Comment:
Do you actually have a use case for this? It seems redundant with `options_mut()`, and I think I'd personally prefer only to expose `options_mut()` instead of having both.

Reply:
Yeah, I wanted to get your take on this actually. It is redundant, but so is `set_option` once we have `options_mut`. The only reason I added `remove_` was so there was some symmetry in the API.

It doesn't matter to me either way though, as long as there's *some* way to accomplish a removal.

Reply:
Yeah, adding `#[deprecated]` on `set_option()` sounds like a plan to me.